### DISCUSSION TITLE
timeout

### DISCUSSION THREAD

**Discussion by RichardWright** [mikro-orm/mikro-orm#3483] [General]
Hi!

Is it possible to set a query timeout (when using a repository) and have it throw?

Thanks

**Comment by B4nan** [comment]
Nope, looks like knex does have support for this for mysql/postgres (the docs are a bit ambiguos, maybe only cancelling is supported in mysql/pg and the general timeout works for all drivers). You could probably use that, I dont mind exposing it to our QB.

https://knexjs.org/guide/query-builder.html#timeout

You could use the QB as usual, and in the end instead of executing it, turn it into knex instance and apply the timeout:

https://mikro-orm.io/docs/query-builder#using-knexjs

(I see the docs mention execution via `em.getConnection().execute(knex)`, in current versions you have `em.execute()` available which should be preferred.)
